packages/wallets/src/evm/connectors/embedded-wallet/index.ts

```diff
@@ -0,0 +1,204 @@
+        clientId: this.options.clientId,
+        chain: "Ethereum",
+        styles: this.options.styles,
+      });
+    }
+    return this.#embeddedWalletSdk;
+  }
+
+  async connect(options?: { email?: string; chainId?: number }) {
+    const thirdwebSDK = await this.getEmbeddedWalletSDK();
+    if (!thirdwebSDK) {
+      throw new Error("Paper SDK not initialized");
+    }
+    const user = await thirdwebSDK.getUser();
+    switch (user.status) {
+      case UserStatus.LOGGED_OUT: {
+        let authResult: AuthLoginReturnType;
+
+        if (options?.email) {
+          authResult = await thirdwebSDK.auth.loginWithEmailOtp({
+            email: options.email,
+          });
+        } else {
+          authResult = await thirdwebSDK.auth.loginWithModal();
+        }
+        this.user = authResult.user;
+        break;
+      }
+      case UserStatus.LOGGED_IN_WALLET_INITIALIZED: {
+        this.user = user;
+        break;
+      }
+    }
+    if (!this.user) {
+      throw new Error("Error connecting User");
+    }
+
+    if (options?.chainId) {
+      this.switchChain(options.chainId);
+    }
+
+    this.setupListeners();
+    return this.getAddress();
+  }
+
+  async disconnect(): Promise<void> {
+    const paper = await this.#embeddedWalletSdk;
+    await paper?.auth.logout();
+    this.#signer = undefined;
+    this.user = null;
+  }
+
+  async getAddress(): Promise<string> {
+    const signer = await this.getSigner();
+    return signer.getAddress();
+  }
+
+  async isConnected(): Promise<boolean> {
+    try {
+      const addr = await this.getAddress();
+      return !!addr;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async getProvider(): Promise<providers.Provider> {
+    const signer = await this.getSigner();
+    if (!signer.provider) {
+      throw new Error("Provider not found");
+    }
+    return signer.provider;
+  }
+
+  public async getSigner(): Promise<Signer> {
+    if (this.#signer) {
+      return this.#signer;
+    }
+
+    if (!this.user) {
+      const embeddedWalletSdk = await this.getEmbeddedWalletSDK();
+      const user = await embeddedWalletSdk.getUser();
+      switch (user.status) {
+        case UserStatus.LOGGED_IN_WALLET_INITIALIZED: {
+          this.user = user;
+          break;
+        }
+      }
+    }
+
+    const signer = await this.user?.wallet.getEthersJsSigner({
+      rpcEndpoint: this.options.chain.rpc[0],
+    });
+
+    if (!signer) {
+      throw new Error("Signer not found");
+    }
+
+    this.#signer = signer;
+
+    return signer;
+  }
+
+  async isAuthorized(): Promise<boolean> {
+    return false;
+  }
+
+  async switchChain(chainId: number): Promise<void> {
+    const chain = this.options.chains.find((c) => c.chainId === chainId);
+    if (!chain) {
+      throw new Error("Chain not configured");
+    }
+
+    // update chain in wallet
+    await this.user?.wallet.setChain({ chain: "Ethereum" }); // just pass Ethereum no matter what chain we are going to connect
```

Comment:
Does the `chain` in the underlying embedded wallet basically not matter anymore?